src/java/main/org/apache/zookeeper/server/NIOServerCnxn.java

```diff
@@ -479,7 +479,7 @@ private boolean checkFourLetterWord(final SelectionKey k, final int len)
         // We take advantage of the limited size of the length to look
         // for cmds. They are all 4-bytes which fits inside of an int
         String cmd = FourLetterCommands.getCmdMapView().get(len);
-        if (cmd == null) {
+        if (cmd == null || !FourLetterCommands.getWhiteListedCmdView().contains(cmd)) {
```

Comment:
What do you suggest it can be done here?

Maybe throw an exception if ``!FourLetterCommands.getWhiteListedCmdView().contains(cmd)`` is ``true`` and get it in the callee?